src/Machine.js

```diff
@@ -584,6 +614,192 @@ class Machine {
+    }
+  }
+
+  /**
+   * Create all missing folders in the current path
+   * @param {string} dirPath path to directory
+   * @private
+   */
+  _mkdirpSync(dirPath) {
+    const parts = dirPath.split(/[\\|\/]/);
+
+    // For every part of our path, call our wrapped _mkdirSync()
+    // on the full path until and including that part
+    for (let i = 1; i <= parts.length; i++) {
+      this._mkdirSync(path.join.apply(null, parts.slice(0, i)));
+    }
+  }
+
+  /**
+   * Create file with fileName path
+   * @param {string} fileName path to file
+   * @param {string} fileContent Content of the file
+   * @private
+   */
+  _createFile(fileName, fileContent) {
+    try {
+      fs.writeFileSync(fileName, fileContent);
+    } catch (err) {
+       this.logger.error(err);
+    }
+  }
+
+  /**
+   * @typedef {Object} NormalizedPath
+   * @property {string} fileName
+   * @property {string} dirPath
+   */
+
+  /**
+   * Transform url or github link to path and filename
+   * It is important, that path and filename are unique,
+   * because collision can break the build
+   * @param {string} link link to the file
+   * @return {NormalizedPath} folder and name, where cache file can be found
+   * @private
+   */
+ _normalizePath(link) {
+    const ghRes = GithubReader._parse(link);
+    if (ghRes !== false) {
+      return this._normalizeGithubPath(ghRes);
+    }
+    if (new HttpReader().supports(link)) { // CHANGE THIS
+      return this._normalizeHttpPath(link);
+    }
+  }
+
+  /**
+   * Transform github link to path and filename
+   * @param {string} link link to the file
+   * @return {NormalizedPath} folder and name, where cache file can be found
+   * @private
+   */
+ _normalizeHttpPath(httpLink) {
+    // parse url parts
+    const parsedUrl = (/^((http[s]?):\/)?\/?([^:\/\s]+)((\/[\w\-\.]+)*\/)([\w\-\.]+[^#?\s]+)(.*)?(#[\w\-]+)?$/.exec(httpLink));
+    const domen = parsedUrl[3].split('.'); // it is web-site name
+    // create new path from url
+    const newPath = this.cacheDir + HTTP_DIR + '/' + domen.filter((elem) => elem != 'www').reverse().join('/') +  (parsedUrl[4] ? parsedUrl[4] : '');
+    const fileName = parsedUrl[6];
+    return { 'dirPath' : newPath,
+             'fileName' : fileName};
+  }
+
+  /**
+   * Transform url link to path and filename
+   * @param {{user, repo, path, ref}} ghRes github parsed link to the file
+   * @return {NormalizedPath} folder and name, where cache file can be found
+   * @private
+   */
+ _normalizeGithubPath(ghRes) {
+    // find, where fileName starts
+    const i = ghRes.path.lastIndexOf('/');
+    let newPath;
+    let fileName;
+    // check case, when filename goes after user and repo
+    if (i != -1) {
+      newPath = ghRes.path.substr(0, i);
+      fileName = ghRes.path.substr(i + 1);
+    }
+    return {
+      'dirPath' : this.cacheDir + GITHUB_DIR + '/' + ghRes.user + '/' + ghRes.repo + (ghRes.ref ? '/' + ghRes.ref : '') + (i != -1 ? '/' + newPath : ''),
```

Comment:
Not sure that's the right way to build path. Platform specific routines should be used https://nodejs.org/api/path.html